src/group_theory/iwasawa.lean

```diff
@@ -0,0 +1,655 @@
+-- import group_theory.subgroup.basic
+import group_theory.subgroup.pointwise
+import group_theory.coset
+import group_theory.quotient_group
+import group_theory.abelianization
+import group_theory.group_action.defs
+import group_theory.group_action.basic
+import group_theory.group_action.group
+import group_theory.group_action.conj_act
+
+
+/- 
+# Iwasawa criterion and simplicity of some groups
+
+Reference : https://www.math.ens.fr/~vanhaecke/tdalg2019/Correction_Iwasawa.pdf
+
+## Iwasawa criterion
+
+Let G be a group acting on a set X.
+
+**Definition**. The action is *transitive* if there is x:X whose orbit is X
+
+**Definition.** The action is *primitive* if it is transitive,
+and if all stabilizers G_x are maximal subgroups.
+
+**Lemma.** If the action is 2-transitive, then it is primitive.
+
+**Theorem (Iwasawa criterion).** Let G be a group with a primitive action on a set X.
```

Comment:
Thanks for this module docstring at the top of the file!
Could you please cross-link to the Lean code below, by adding the name of Lean theorems/defs (in backticks) to this list of thms/defs? The generated html documentation will automatically turn those into hyperlinks and people who open this file will quickly know the Lean name of the main result.